// Author: Kun Ren <kun@cs.yale.edu>
//
syntax = "proto2";
message TpccAction {  // Actually a namespace!

// Types of actions that KVStore can interpret.
enum Type {
  NEW_ORDER = 0;
  PAYMENT = 1;
  ORDER_STATUS = 2;
  DELIVERY = 3;
  STOCK_LEVEL = 4;
};

/////////////////////////
// NEW_ORDER
message NewOrderInput {
  repeated bytes keys = 1;
}
// (no output type)

/////////////////////////
// PAYMENT
message PaymentInput {
  repeated bytes keys = 1;
}
// (no output type)

}  // message MicrobenchmarkAction
